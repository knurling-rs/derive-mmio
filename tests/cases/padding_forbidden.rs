#[derive(derive_mmio::Mmio)]
#[repr(C)]
struct Uart {
    // this is read-only by default
    data: u32,
    // this will introduce padding, which will fail the compilation
    blub: u16,
    // this will introduce padding, which will fail the compilation
    _reserved2: u8,
}

fn main() {}
