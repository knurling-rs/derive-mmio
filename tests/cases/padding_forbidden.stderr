error[E0308]: mismatched types
 --> tests/cases/padding_forbidden.rs:1:10
  |
1 | #[derive(derive_mmio::Mmio)]
  |          ^^^^^^^^^^^^^^^^^ expected an array with a size of 7, found one with a size of 8
  |
  = note: this error originates in the derive macro `derive_mmio::Mmio` (in Nightly builds, run with -Z macro-backtrace for more info)
